aircraft/landing_gear.py

```diff
@@ -0,0 +1,270 @@
+
+        # Constants
+        d_nac     = Variable('d_{nacelle}', 2, 'm', 'Nacelle diameter')
+        E         = Variable('E', 205, 'GPa',
+                             'Modulus of elasticity, 4340 steel') # ASM Matweb
+        eta_s     = Variable('\\eta_s', 0.8, '-', 'Shock absorber efficiency')
+        eta_t     = Variable('\\eta_t', 0.47, '-',
+                             'Efficiency of tire in shock absorption')
+        g         = Variable('g', 9.81, 'm/s^2', 'Gravitational acceleration')
+        h_nac     = Variable('h_{nacelle}', 0.5, 'm', 'Min. nacelle clearance')
+        hhold     = Variable('h_{hold}', 1, 'm', 'Hold height')
+        K         = Variable('K', 2, '-', 'Column effective length factor')
+        lam       = Variable('\\lambda', 2.5, '-',
+                             'Ratio of max to static load') # Torenbeek p360
+        n_mg      = Variable('n_{mg}', 2, '-', 'Number of main gear struts')
+        N_s       = Variable('N_s', 2, '-', 'Factor of safety')
+        nwps      = Variable('n_{wps}', 2, '-', 'Number of wheels per strut')
+        p_oleo    = Variable('p_{oleo}', 1800, 'lbf/in^2', 'Oleo pressure')
+#        p_t       = Variable('p_t', 170, 'lbf/in^2', 'Tyre pressure')
+        rho_st    = Variable('\\rho_{st}', 7850, 'kg/m^3',
+                             'Density of 4340 Steel') # ASM Matweb
+        sig_y_c   = Variable('\\sigma_{y_c}', 470E6, 'Pa',
+                             'Compressive yield strength 4340 steel ') #  AZOM
+        tan_15    = Variable('\\tan(\\phi_{min})', np.tan(15*np.pi/180), '-',
+                             'Lower bound on phi')
+        tan_63    = Variable('\\tan(\\psi_{max})', np.tan(63*np.pi/180), '-',
+                             'Upper bound on psi')
+        tangam    = Variable('\\tan(\\gamma)', np.tan(5*np.pi/180), '-',
+                             'Tangent of dihedral angle') 
+        tan_th_TO = Variable('\\tan(\\theta_{TO})', np.tan(15*np.pi/180), '-',
+                             'Takeoff pitch angle')
+        w         = Variable('w', 10, 'ft/s', 'Ultimate velocity of descent')
+        W_0       = Variable('W_0', 82000*9.81, 'N', # 737 Max airport doc
+                             'Weight of aircraft excluding landing gear')
+        xcg0      = Variable('x_{CG_0}', 18, 'm', 'x-location of CG excl. LG')
+        x_upswp   = Variable('x_{up}', 28, 'm', 'Fuselage upsweep point')
+        y_eng     = Variable('y_{eng}', 4.83, 'm', 'Spanwise loc. of engines')
+        z_CG_0    = Variable('z_{CG}', 2, 'm',
+                             'CG height relative to bottom of fuselage')
+        zwing     = Variable('z_{wing}', 0.5, 'm',
+                             'Height of wing relative to base of fuselage')
+
+        with SignomialsEnabled():
+
+            objective = W_lg
+
+            constraints = [
+                           # Track and Base geometry definitions
+                           l_n + zwing + y_m*tangam >= l_m, # [SP]
```

Comment:
yet another example of a Signomial Constraint where as a modeler I know a good initial guess:

a reasonable initial monomial approximation of `l_n + zwing + y_m*tangam` is `l_n`.